src/service/viewer-impl.js

```diff
@@ -183,6 +184,9 @@ export class Viewer {
     /** @const @private {!Object<string, string>} */
     this.params_ = {};
 
+    /** @const @private {!Object<string, string>} */
+    this.hashParams_ = {};
```

Comment:
Please add a comment describing this new property vis-a-vis `params_`.

E.g. `Subset of this.params_ that only contains parameters in the URL hash e.g. "#foo=bar".`